### PR TITLE
build: add release workflow

### LINKED ISSUE #55: build: generate node release
Automatically build and attach node binary to published release.

### PR DESCRIPTION
**Pull request by evilrobot-01**
Closes #55

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  push:
+    branches:
+      - 'frank/build-release'
+
+jobs:
+  build-binary:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        cpu: [ "", "skylake" ]
```
Comment (peterwht) [inline .github/workflows/release.yml:20]: per the failing CI, probably easiest to remove "skylake" from here and just build on regular CPU.

Comment (peterwht) [inline .github/workflows/release.yml:20]:
```suggestion
        cpu: [ ""]
```

Per the failing CI, probably easiest to remove "skylake" and just build for regular CPU.

Comment (evilrobot-01) [reply .github/workflows/release.yml:20]: Nah, its something that v4 of the upload artifacts breaks, so downgrading.

https://github.com/actions/upload-artifact/issues/478

**Review by peterwht — APPROVED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  release:
+    types: [ published ]
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: ref to build binary from
+        required: false
+  push:
+    branches:
+      - 'frank/build-release'
```
Comment (peterwht) [inline .github/workflows/release.yml:13]: Presumably, this is just here for initial testing?

Comment (evilrobot-01) [reply .github/workflows/release.yml:13]: Yeah, it's the only way to have the workflow appear in GH UI pre-merge to main.

**Review by al3mart — APPROVED** [review]

**Review by al3mart — APPROVED** [review]
:+1:

File: .github/workflows/release.yml (hunk 2 above)
Comment (evilrobot-01) [reply .github/workflows/release.yml]: Resolved in latest commit

**Review by Daanvdplas — APPROVED** [review]

**Pull request merged**